genesis/src/lockups.rs

```diff
@@ -0,0 +1,160 @@
+//! lockups generator
+use solana_sdk::{clock::Epoch, epoch_schedule::EpochSchedule, timing::years_as_slots};
+use std::time::Duration;
+
+#[derive(Debug, Default, Clone)]
+pub struct Lockups {
+    /// where in iteration progress I am
+    expiration: usize,
+    /// number of expirations post cliff
+    expirations: usize,
+    /// fractionage offered to f() last time
+    prev_fraction: f64,
+    /// portion of total available at cliff
+    cliff_fraction: f64,
```

Comment:
Looks like you switched naming conventions mid-struct.  `cliff_` became a namespace. But then there's `epoch` and `epochs` where I thought both would have been `epochs`. But then the type of both `epoch` and `epochs` is `Epoch` - I'm lost.

Reply:
yeah: the first cliff is zero-based, the increments are increments from there